lib/hooks/core/hook-http.js

```diff
@@ -43,19 +43,19 @@ function requestWrap(request) {
 
     var root = cls.getRootContext();
 
+    options = (typeof options === 'string') ?
+      url.parse(options) : clone(options);
+    options.headers = options.headers || {};
+    var uri = uriFromOptions(options);
```

Comment:
The logic required to get the Uri is quite small. I've added it inside the `if (!root) {` case.